vendor/bat-native-ledger/src/bat/ledger/internal/gemini/gemini.cc

```diff
@@ -0,0 +1,289 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include <utility>
+
+#include "base/guid.h"
+#include "base/strings/string_number_conversions.h"
+#include "base/strings/stringprintf.h"
+#include "bat/ledger/global_constants.h"
+#include "bat/ledger/internal/common/time_util.h"
+#include "bat/ledger/internal/endpoint/gemini/gemini_server.h"
+#include "bat/ledger/internal/gemini/gemini.h"
+#include "bat/ledger/internal/gemini/gemini_authorization.h"
+#include "bat/ledger/internal/gemini/gemini_transfer.h"
+#include "bat/ledger/internal/gemini/gemini_util.h"
+#include "bat/ledger/internal/gemini/gemini_wallet.h"
+#include "bat/ledger/internal/ledger_impl.h"
+#include "bat/ledger/internal/logging/event_log_keys.h"
+#include "bat/ledger/internal/state/state_keys.h"
+#include "bat/ledger/internal/wallet/wallet_util.h"
+#include "brave_base/random.h"
+
+using std::placeholders::_1;
+using std::placeholders::_2;
+using std::placeholders::_3;
+
+namespace {
+const char kFeeMessage[] =
+    "5% transaction fee collected by Brave Software International";
+}  // namespace
+
+namespace ledger {
+namespace gemini {
+
+Gemini::Gemini(LedgerImpl* ledger)
+    : transfer_(std::make_unique<GeminiTransfer>(ledger)),
+      authorization_(std::make_unique<GeminiAuthorization>(ledger)),
+      wallet_(std::make_unique<GeminiWallet>(ledger)),
+      gemini_server_(std::make_unique<endpoint::GeminiServer>(ledger)),
+      ledger_(ledger) {}
+
+Gemini::~Gemini() = default;
+
+void Gemini::Initialize() {
+  auto wallet = GetWallet();
+  if (!wallet) {
+    return;
+  }
+
+  for (const auto& value : wallet->fees) {
+    StartTransferFeeTimer(value.first, 1);
+  }
+}
+
+void Gemini::StartContribution(const std::string& contribution_id,
+                               type::ServerPublisherInfoPtr info,
+                               const double amount,
+                               ledger::ResultCallback callback) {
+  if (!info) {
+    BLOG(0, "Publisher info is null");
+    ContributionCompleted(type::Result::LEDGER_ERROR, "", contribution_id,
+                          amount, "", callback);
+    return;
+  }
+
+  const double fee = (amount * 1.05) - amount;
```

Comment:
declare `1.05` as const in an empty namespace

Reply:
fixed: https://github.com/brave/brave-core/pull/8566/files#diff-4bd796bc3bb28fd3579feb8625aa26eb4100fe6c46162c88ec36ec54f3b74f00R32